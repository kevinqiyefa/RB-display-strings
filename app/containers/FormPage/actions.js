/*
 *
 * HomePage actions
 *
 */
import {
  POST_STRING_REQUEST,
  POST_STRING_SUCCESS,
  POST_STRING_FAILURE,
} from './constants';

export function postStringRequest(string) {
  return {
    type: POST_STRING_REQUEST,
    string,
  };
}

export function postStringSuccess() {
  return {
    type: POST_STRING_SUCCESS,
  };
}

export function postStringFailure(err) {
  return {
    type: POST_STRING_FAILURE,
    err,
  };
}
