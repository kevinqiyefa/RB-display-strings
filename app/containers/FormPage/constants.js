/*
 *
 * FormPage constants
 *
 */
export const POST_STRING_REQUEST = 'POST_STRING_REQUEST';
export const POST_STRING_SUCCESS = 'POST_STRING_SUCCESS';
export const POST_STRING_FAILURE = 'POST_STRING_FAILURE';
