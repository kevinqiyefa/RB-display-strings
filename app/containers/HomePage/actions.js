/*
 *
 * HomePage actions
 *
 */
import {
  FETCH_STRING_REQUEST,
  FETCH_STRING_SUCCESS,
  FETCH_STRING_FAILURE,
} from './constants';

export function fetchStringRequest() {
  return {
    type: FETCH_STRING_REQUEST,
  };
}

export function fetchStringSuccess(strings) {
  return {
    type: FETCH_STRING_SUCCESS,
    strings,
  };
}

export function fetchStringFailure(err) {
  return {
    type: FETCH_STRING_FAILURE,
    err,
  };
}
