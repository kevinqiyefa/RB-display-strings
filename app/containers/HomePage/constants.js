/*
 *
 * HomePage constants
 *
 */
export const FETCH_STRING_REQUEST = 'FETCH_STRING_REQUEST';
export const FETCH_STRING_SUCCESS = 'FETCH_STRING_SUCCESS';
export const FETCH_STRING_FAILURE = 'FETCH_STRING_FAILURE';
