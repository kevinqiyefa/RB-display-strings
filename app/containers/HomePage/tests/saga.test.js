/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';
import { FETCH_STRING_REQUEST } from '../constants';
import { fetchStringSuccess, fetchStringFailure } from '../actions';
import homePageSaga, { getStrings } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getStrings Saga', () => {
  let getStringsGenerator;

  beforeEach(() => {
    getStringsGenerator = getStrings();

    const callDescriptor = getStringsGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the fetchStringSuccess action if it requests the data successfully', () => {
    const response = [
      {
        id: 1,
        string: 'test1',
      },
      {
        id: 2,
        string: 'test2',
      },
    ];
    const putDescriptor = getStringsGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(fetchStringSuccess(response)));
  });

  it('should call the fetchStringFailure action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getStringsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(fetchStringFailure(response)));
  });
});

describe('homePageSaga Saga', () => {
  const homePage = homePageSaga();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = homePage.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(FETCH_STRING_REQUEST, getStrings),
    );
  });
});
