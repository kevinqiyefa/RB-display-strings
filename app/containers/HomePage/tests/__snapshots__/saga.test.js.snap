// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStrings Saga should call the fetchStringFailure action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getStrings Saga should dispatch the fetchStringSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
